shark_turbine/kernel/wave/expansion.py

```diff
@@ -208,14 +214,40 @@ def expand_graph(
                 dim: val for dim, val in zip(dim_scaling.keys(), dim_combination)
             }
             logger.debug(f"Starting expansion at leaf:{node} in dims:{expand_dims}")
-            _expand_node(node, trace, expand_dims, dim_scaling, expansion_context)
+            _expand_node(
+                node, trace, expand_dims, dim_scaling, dim_tile_size, expansion_context
+            )
+
+    # Replace symbolic tile sizes with actual tile sizes.
+    def replace_index(node: fx.Node) -> bool:
+        if node.index is not None:
+            for dim in node.index:
+                node.index[dim].start = node.index[dim].start.subs(tile_size_mapping)
+        return False
+
+    trace.walk(replace_index)
+
+
+def update_index(
+    restricted_dims: dict[IndexSymbol, int],
+    dim_tile_size: dict[IndexSymbol, int],
+    new_node: fx.Node,
+) -> None:
+    for dim, index in restricted_dims.items():
+        if new_node.fx_node.index is not None and dim in new_node.fx_node.index:
+            for tile_size in dim_tile_size.values():
+                new_node.fx_node.index[dim].start = new_node.fx_node.index[
+                    dim
+                ].start.subs({tile_size: 0})
```

Comment:
I am not sure if I understand this correctly, are we subbing in the value of the dim_tile_size(i.e the int values inside vector shape) by 0?

Reply:
See other comment, but yes, we are erasing all previous instances of tile size to avoid overcounting and then applying the index offset.

Reply:
I see I see, but wouldn't the index still be symbolic at this stage, since we only call `replace_index` after we finish expanding all the nodes and there won't be any more calls to `update_index`?

Reply:
Yes that's right. The index will still be symbolic. So it would be something like `symbolic_tile_size * 2` and then we would zero it out and then set it to say `symbolic_tile_size * 3`. (If we didn't zero it out, we would end up with `symbolic_tile_size * 2 + symbolic_tile_size * 3`.

Reply:
should we not be iterating over keys instead of values then? 

as in `for tile_size in dim_tile_size.keys():` instead of `for tile_size in dim_tile_size.values():`?

Reply:
So dim_tile_size is a dictionary from dimension to tile size and looks something like {M: TILE_M, N: TILE_N}.
Our original expression looks something like: index = 2 * TILE_N, 3 * TILE_M
so we want to iterate over the values (TILE_M, TILE_N) and set those to zero to get index = 0 and then update the index.
Does that make sense?

Reply:
removed this part of the code.